### PR TITLE
Add log output to Windows Event Log

### PR DESCRIPTION
**Pull request by andrewkroh**
By setting `logging.to_eventlog: true` all log output will be written
to the Application log. The source name will be name of the Beat.

### CODE REVIEW

**Comment by andrewkroh** [comment]
I rebased this to fix the CHANGELOG merge conflict.

File: auditbeat/auditbeat.reference.yml [hunk 1]
```diff
@@ -845,7 +844,10 @@ setup.kibana:
 #logging.selectors: [ ]
 
 # Send all logging output to syslog. The default is false.
-#logging.to_syslog: true
+#logging.to_syslog: false
+
+# Send all logging output to Windows Event Logs. The default is false.
+#logging.to_eventlog: false
```
Comment (ruflin) [inline auditbeat/auditbeat.reference.yml]: Not a big fan of naming we have for historical reason and I'm thinking we should change it for 7.0. Perhaps we could already follow here a new scheme similar to what we have in other places:

`logging.eventlog.enabled`. Like this we can have specific logging options under the namespace.

File: libbeat/logp/encoding.go [hunk 2]
```diff
@@ -6,6 +6,21 @@ import (
 	"go.uber.org/zap/zapcore"
 )
 
+var baseEncodingConfig = zapcore.EncoderConfig{
```
Comment (ruflin) [inline libbeat/logp/encoding.go]: +1 on the cleanup

File: libbeat/logp/eventlog_windows.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap/zapcore"
+	"golang.org/x/sys/windows/svc/eventlog"
+)
+
+const (
+	eventID  = 100
```
Comment (ruflin) [inline libbeat/logp/eventlog_windows.go]: Does `100` have a special meaning in windows event logs?

File: libbeat/logp/eventlog_windows.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	supports = eventlog.Warning | eventlog.Info | eventlog.Info
```
Comment (ruflin) [inline libbeat/logp/eventlog_windows.go]: Why does `eventlog.Info` show up twice? Intentional or should it be Debug?

File: libbeat/logp/syslog_unix.go [hunk 5]
```diff
@@ -53,7 +53,7 @@ func (c *syslogCore) Write(entry zapcore.Entry, fields []zapcore.Field) error {
 	}
 
 	// Console encoder writes tabs which don't render nicely with syslog.
-	replaceTabsWithSpaces(buffer.Bytes(), 2)
+	replaceTabsWithSpaces(buffer.Bytes(), 4)
```
Comment (ruflin) [inline libbeat/logp/syslog_unix.go]: Is that change related to this PR?

File: auditbeat/auditbeat.reference.yml (hunk 1 above)
Comment (andrewkroh) [reply auditbeat/auditbeat.reference.yml]: I am 100% on board with changing how logging is configured for 7.0. I did it this way to be consistent. I will open an issue to track things that I'd like to "break" for 7.0 (I have a list 😄 ).

File: libbeat/logp/eventlog_windows.go (hunk 3 above)
Comment (andrewkroh) [reply libbeat/logp/eventlog_windows.go]: No, it's completely arbitrary. The value needs to be between 1 and 1000. Most of the original log events have three digit IDs so I just chose 100.

File: libbeat/logp/eventlog_windows.go (hunk 4 above)
Comment (andrewkroh) [reply libbeat/logp/eventlog_windows.go]: That should be `Error`. Will push a fix.

File: libbeat/logp/syslog_unix.go (hunk 5 above)
Comment (andrewkroh) [reply libbeat/logp/syslog_unix.go]: It's not related. I just realized that there can be up to 4 tabs.

File: auditbeat/auditbeat.reference.yml (hunk 1 above)
Comment (ruflin) [reply auditbeat/auditbeat.reference.yml]: It should be possible to already do the "new" scheme here as so far it's just a name. More thinking then we have to break one option less for 7 :-) But not sure if we are already settle on if that should be new scheme.

+1 on the issue.

Comment (andrewkroh) [reply auditbeat/auditbeat.reference.yml]: Well the scheme I had in mind kind of conflicts. I was thinking of something like.

```
logging.output:
- type: file
   level: info
   json: true
- type: eventlog
  level: warn
```

Comment (ruflin) [reply auditbeat/auditbeat.reference.yml]: got it, then lets keep it as is in your PR and break it later so we don't need to come to a conclusion now.

I like your format as it gives max flexibility. I assume you have in mind that more then 1 log output can be enabled?

Comment (andrewkroh) [reply auditbeat/auditbeat.reference.yml]: Yeah, more than one output and possibly more than one instance of an output type (not 100% sure yet; requires more thought).

**Comment by andrewkroh** [comment]
I just did a manual squash to keep the vendor changes separate from the code. Now you can do a rebase and merge. Thanks
